Handle builds without C++

libxmlrpc nicely handles the fact of being built without C++ support,
except for one location, fixed by this patch.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/lib/util/Makefile
===================================================================
--- a/lib/util/Makefile
+++ b/lib/util/Makefile
@@ -41,11 +41,14 @@
 LIBOBJS = \
   casprintf.o \
   cmdline_parser.o \
-  cmdline_parser_cpp.o \
   getoptx.o \
   string_parser.o \
   stripcaseeq.o \
 
+ifeq ($(ENABLE_CPLUSPLUS),yes)
+LIBOBJS += cmdline_parser_cpp.o
+endif
+
 .PHONY: all
 all: $(LIBOBJS)
 
